### ISSUE TITLE
isStarted() returns true even if agent is not active

### ISSUE THREAD

**Issue by sometimesalready** [elastic/apm-agent-nodejs#1109]
**Describe the bug**
isStarted() returns true even if agent is not active

**To Reproduce**
```javascript
	const apm = require('elastic-apm-node').start({ active: false, ...});
	if (apm.isStarted()) { 
		apm.captureError(new Error(), err => console.error(err));
	}
```
Error logged: `Error: cannot capture error before agent is started`

**Expected behavior**
Expect to use isStarted() as a guard to run `captureError` method

**Environment (please complete the following information)**
- OS: mac-os
- Node.js version: v8.11.2
- Agent version: lastic-apm-node@2.11.5

**How are you starting the agent? (please tick one of the boxes)**
- [x] Calling `agent.start()` directly (e.g. `require('elastic-apm-node').start(...)`)
- [ ] Requiring `elastic-apm-node/start` from within the source code
- [ ] Starting node with `-r elastic-apm-node/start`

**Comment by watson** [comment]
Ah, I see this is quite confusing. Technically all that we're checking inside `isStarted()` is if `start()` has already been called. Not if the agent started successfully as a result of calling `start()`. But I agree that's very confusing.

We should either update the docs to be more explicit about this or change the behavior of the function as you suggest.

What's your use-case for calling `isStarted()`? You shouldn't need to guard any API calls like you do in your example even if the agent isn't active. If you find that you need to do that we have a bug.

**Comment by sometimesalready** [comment]
I need a method to check if I can call `captureError` method safely.

**Comment by alexanderPrytula** [comment]
Could you please expose something like
```
isActive() {
    return this._conf.active;
}
```
to legitimize the usage to that config flag

**Comment by watson** [comment]
@sometimesalready it's safe to ignore the `cannot capture error before agent is started` error returned in the callback in case the agent is inactive. But maybe we shouldn't return an error at all in this case 🤔 Would it solve your problem if the callback wasn't called with an error in this case? Or do you have another reason why you want to encapsulate the call to `captureError()` inside an if-statement?

@alexanderPrytula yeah it might be good to expose the state of that config option anyway. We're [in the process](https://github.com/elastic/apm/issues/76#issuecomment-495137879) of rethinking the different states that the agent can be in when starting up to better align with the other agents

**Comment by sometimesalready** [comment]
yes, it will solve my problem if the error will be not passed in a callback in this scenario cause I promisify `captureError` like this:
```javascript
captureErrorPromisified = util.promisify(apm.captureError.bind(apm));
```
and use `captureErrorPromisified` without checking if it was active or not

**Comment by watson** [comment]
@sometimesalready Ah yes good point - I didn't think of that. We need to ensure it doesn't return an error in this case. Maybe we should also consider only having `isStarted()` return `true` if the agent is actually active - but that's another issue. If not we should at least update the docs to better specify what to expect

**Comment by watson** [comment]
Note to self: If we changed the behavior of `isStarted()` to only return `true` if the agent is active, that could be considered a breaking change.

The way to implement it probably be something like this:

```diff
diff --git a/lib/agent.js b/lib/agent.js
index 1c5d3035..80065ab0 100644
--- a/lib/agent.js
+++ b/lib/agent.js
@@ -114,11 +114,11 @@ Agent.prototype._config = function (opts) {
 }
 
 Agent.prototype.isStarted = function () {
-  return global[symbols.agentInitialized]
+  return global[symbols.agentInitialized] && this._conf.active
 }
 
 Agent.prototype.start = function (opts) {
-  if (this.isStarted()) throw new Error('Do not call .start() more than once')
+  if (global[symbols.agentInitialized]) throw new Error('Do not call .start() more than once')
   global[symbols.agentInitialized] = true
 
   this._config(opts)
```

But then this example code - while mostly theoretical - would suddenly throw because if the throw on the first line of the `start()` function:

```js
const agent = require('elastic-apm-node').start({ active: false })

// ...then later:
if (!agent.isStarted()) agent.start({ active: true }) // will throw
```

**Comment by jiri-vyc** [comment]
Any progress on this?

Now it is not possible to check if the agent is in fact started and really active, as `isStarted()` returns true even if it's inactive.

+1 for exposing `isActive()` on the agent
